bench: refactor random value generation in `math/base/special/cabs*`

saurabhraghuvanshii opened the pull request:
Resolves #none

## Description

> What is the purpose of this pull request?

This pull request:

-   refactor random number generation

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

saurabhraghuvanshii committed:
bench: refactor random generation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii commented:
@anandkaranubc I need help , I am little bit confuse on refactor these file in `benchmark.c` , If what I do is right then I will add more file . Please review it .

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cabs/benchmark/c/native/benchmark.c:
```diff
@@ -92,20 +92,23 @@ static double rand_double( void ) {
-		re = ( 1000.0*rand_double() ) - 500.0;
-		im = ( 1000.0*rand_double() ) - 500.0;
-		z = stdlib_complex128( re, im );
+		z = stdlib_complex128( re[ i % 100 ], im[ i % 100 ] );
```
```suggestion
		z = stdlib_complex128( re[ i%100 ], im[ i%100 ] );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cabs2/benchmark/c/benchmark.c:
```diff
@@ -99,19 +99,22 @@ double cabs2( double complex z ) {
-		re = ( 1000.0*rand_double() ) - 500.0;
-		im = ( 1000.0*rand_double() ) - 500.0;
-		z = re + im*I;
+		z = re[ i%100] + im[ i%100 ]*I;
```
```suggestion
		z = re[ i%100 ] + im[ i%100 ]*I;
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cabs2/benchmark/c/native/benchmark.c:
```diff
@@ -92,20 +92,23 @@ static double rand_double( void ) {
-		re = ( 1000.0*rand_double() ) - 500.0;
-		im = ( 1000.0*rand_double() ) - 500.0;
-		z = stdlib_complex128( re, im );
+		z = stdlib_complex128( re[ i % 100 ], im[ i % 100 ] );
```
```suggestion
		z = stdlib_complex128( re[ i%100 ], im[ i%100 ] );
```

anandkaranubc reviewed (changes requested):
Some minor spacing issues, but the approach looks fine to me. Can you also confirm that by running the benchmarks locally?

saurabhraghuvanshii commented:
@anandkaranubc , It is working I tested locally , thanks

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into random-Ca

saurabhraghuvanshii committed:
chore: refactor cabsf and cabs2f

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii commented:
closing, merge conflicts thanks
